### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by arnoldlaoang**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -17,35 +17,104 @@
...
 
+describe("namer", () => {
+  it("take in an array of objects and returns an array with a sentence about each person with their name capitalized.", () => {
+    expect(namer(people)).toEqual([
+      "Ford Prefect is a hitchhiker",
+      "Zaphod Beeblebrox is president of the galaxy",
+      "Arthur Dent is a radio employee",
+    ]);
+  });
+});
```
Comment (SunkissedQueen) [inline code-challenges.test.js:34]: good structure

File: code-challenges.test.js [hunk 2]
```diff
@@ -17,35 +17,104 @@
...
+    var names = allname.split(" ");
+    var fname = names[0];
+    var lname = names[1];
+    fname = fname[0].toUpperCase() + fname.slice(1);
+    lname = lname[0].toUpperCase() + lname.slice(1);
+    var fullname = fname + " " + lname;
+    return fullname + " is " + person.occupation;
+  });
+};
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:66]: Great work. Please include the results of your test

File: code-challenges.test.js [hunk 3]
```diff
@@ -17,35 +17,104 @@
...
 // b) Create the function that makes the test pass.
 
+// Pseudocode:
+// Create a function that takes in an a mixed data array
+// Use a higher order function .filter to filter through the returned array
+// Inside the method parameter declare the value as a placeholder
+// Go through the mixed data array by using typeof to get number values to placeholder
+// Use higher order function .map to go through the new array with only number values
+// Inside the method parameter declare the value modulo 3 to the placeholder to get the remainders of the numbers in the array.
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:98]: Concise and clean. Beautiful function

File: code-challenges.test.js [hunk 4]
```diff
@@ -17,35 +17,104 @@
...
+// Pseudocode:
+//Create a function that takes in an array of numbers
+// Use the higher order function .map to go through the returned array
+// In the method use the built in method .pow to iterate on all of the numers with the exponent three times passed to placeholder
+// Use the built in method .reduce to return a single value out of the sum of everything inside the arrays respectively
+
+const adder = (array) => {
+  return array.map((value) => Math.pow(value, 3)).reduce((previousValue, currentValue) => previousValue + currentValue, 0);
+};
```
Comment (SunkissedQueen) [inline code-challenges.test.js:135]: Indentation, testing, pseudocode, and function look great. Please make sure you are documenting the test results before and after creating a function.

File: interview-questions.md [hunk 5]
```diff
@@ -7,44 +8,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+cohort1 = new Cohort()
+Title: Alpha
+Id: 1 (primarykey)
+
+Cohort -> (has_many) Students
+
+Student Model:
+Name: string
+Age: integer
+cohort_id: student_id (foreignkey)
```
Comment (SunkissedQueen) [inline interview-questions.md:34]: Like that you made a diagram. Each table will receive its own unique id. The model with the belongs_to will have a foreign key. In this case, the Student Model will belong to the Cohort Model. The foreign key will be the primary key of the cohort, which is written as cohort_id: integer

File: interview-questions.md [hunk 6]
```diff
@@ -7,44 +8,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Your answer:
+Your answer:
 
-  Researched answer:
+- Thinking from what i remember about the RESTful routes i think it will be the update method because that's when it will passed params to make changes.
 
+Researched answer:
 
+- Through my research I learned that the RESTful routes that must always passed params are request GET, PATCH and DELETE because these actions requires passing params throughout the database after doing the CRUD actions.
```
Comment (SunkissedQueen) [inline interview-questions.md:44]: Please review the section in the syllabus that gives more detail about which restful routes and controller action need params. 
https://github.com/learn-academy-2022-alpha/Syllabus/blob/main/rails/restful-routes-crud.md

File: interview-questions.md [hunk 7]
```diff
@@ -7,44 +8,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
-
-  Researched answer:
+Your answer: The three rails generator command that I can rememeber on top of my head are: rails server that starts server in localhost3000, rails console that allows you to type and interact with commands on the console, and rails new which creates a new rails.
 
+Researched answer:
 
+- Through my research I learned that the three rails generator command are: rails generate controller, rails generate model and rails generate migration. The rails generate controller creates the controller file and everything that has to do with the controller.
```
Comment (SunkissedQueen) [inline interview-questions.md:52]: What does the rails generate model and rails generate migration create?

File: interview-questions.md [hunk 8]
```diff
@@ -7,44 +8,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+action: "POST" location: /students
 
-action: "GET"    location: /students/2  
+- The controller method that would be called by each route is the create method and the action that the create method perform is to be able to add new students to the list.
 
-action: "GET"    location: /students/2/edit    
+action: "GET" location: /students/new
 
-action: "PATCH"  location: /students/2      
+- The controller method that would be called by each route is the new method and the action that the new method perform is to render data after creating new student.
```
Comment (SunkissedQueen) [inline interview-questions.md:67]: What else is unique to `new`?

File: interview-questions.md [hunk 9]
```diff
@@ -7,44 +8,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-action: "DELETE" location: /students/2      
+action: "GET" location: /students/2
 
+- The controller method that would be called by each route is the show method and the action that the show method perform is to show a single student on the list in particular.
+
+action: "GET" location: /students/2/edit
+
+- The controller method that would be called by each route is the edit method and the action that the edit method perform is that it will call out a single student on the list in particular and allows you to go the page and make edits.
```
Comment (SunkissedQueen) [inline interview-questions.md:75]: What else is unique to `edit`?

File: interview-questions.md [hunk 10]
```diff
@@ -7,44 +8,90 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+- The controller method that would be called by each route is the update method and the action that the update method perform is that it will allow allow the edit that's been made and finalize changes about the student on the list in particular.
+
+action: "DELETE" location: /students/2
+
+- The controller method that would be called by each route is the destroy method and the action that the destroy method perform is that it deletes data from the student on the list in particular.
 
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
```
Comment (SunkissedQueen) [inline interview-questions.md:87]: Very thorough user stories.

File: rails-commenting.rb [hunk 11]
```diff
@@ -7,24 +7,33 @@
 # FILE: app/controller/blog_posts_controller.rb
```
Comment (SunkissedQueen) [inline rails-commenting.rb:8]: Great work on this section.

**Review by SunkissedQueen — CHANGES_REQUESTED** [review]
Last assessment, Arnold! The assessments might end but the learning continues. Please include the test results after you created the test and after creating the function. I left notes in other sections for you to consider with your studies. I appreciate the structure of your tests and code blocks. Good work!

File: code-challenges.test.js (hunk 1 above)
Comment (arnoldlaoang) [reply code-challenges.test.js:34]: oh my i'm so sorry "banging my head on the desk!"

**Review by SunkissedQueen — APPROVED** [review]

**Pull request merged**